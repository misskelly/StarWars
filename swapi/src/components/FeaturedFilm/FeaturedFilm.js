import React from 'react';

const FeaturedFilm = ({film}) => {
  return (
    <section className='film-container'>
      <article className='scroll-container'>
      {film.scroll}
      </article>
      <article className='film-info'>
        <h3 className='film-title'>{film.title}</h3>
        <h4 className='release-year'>{film.year}</h4>
      </article>
    </section>
  )
}

export default FeaturedFilm