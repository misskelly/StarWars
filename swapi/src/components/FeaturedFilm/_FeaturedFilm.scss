.film-container {
  grid-area: film;
}

h3, h4, h5, {
  margin: 0;
}
.crawler {
  background: transparent;
}

.film-info {
  text-align: center;
}
.film-title {
  line-height: 1;
  font-family: $alien;
  margin-top: 3rem;
}