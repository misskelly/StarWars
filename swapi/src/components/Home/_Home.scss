
.planets-btn {
  background: #159957;
  background: -webkit-linear-gradient(to right, #159957, #155799);
  background: linear-gradient(to right, #159957, #155799);
}

.vehicles-btn {
  background: rgb(124, 167, 29);
  background: radial-gradient(circle, rgba(124, 167, 29, 1) 15%, rgba(10, 108, 2, 1) 100%);
}

.favs-btn {
  background: rgb(255, 99, 34);
  background: radial-gradient(circle, rgba(255, 99, 34, 1) 0%, rgba(254, 8, 60, 1) 100%);
}

.people-btn {
  background: rgb(255, 216, 0);
  background: linear-gradient(133deg, rgba(255, 216, 0, 1) 0%, rgba(254, 187, 8, 1) 22%, rgba(254, 121, 8, 1) 90%);
}

.home {
  display: grid;
  grid-template-columns: 2fr 3fr 2fr;
  grid-template-rows: auto auto 1fr;
  grid-template-areas:
    "left film right"
    "left film right"
    "left ship right";
}


