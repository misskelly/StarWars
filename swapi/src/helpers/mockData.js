export const mockCleanFilm = {
     scroll: "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy …",
     title: "The Phantom Menace",
     year: "1999"
   }
export const mockDirtyFilm = {
     opening_crawl: "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy …",
     title: "The Phantom Menace",
     release_date: "1999-07-03"
   }


