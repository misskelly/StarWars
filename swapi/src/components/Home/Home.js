  import React from 'react'
  import Ship from '../Ship/Ship'
  import FeaturedFilm from '../FeaturedFilm/FeaturedFilm'
  
  const Home = ({film, updateActiveComponent}) => {
    
    const handleClick = (e) => {
      const category = e.target.id;
      updateActiveComponent(category)
    }
    return (
      <section className='home'>
        <section className='left-btns'>
          <div className='moon'></div>
            <button 
              className='planets-btn planet-btn'
              id='planets' 
              onClick={handleClick}>
              Planets
              </button>
              <button 
                className='people-btn planet-btn'
                id='people' 
                onClick={handleClick}>
              People
              </button>
            </section>

            <FeaturedFilm film={ film }/>
            
            <section className='right-btns'>
              <button 
                className='favs-btn planet-btn'
                id='favs' 
                onClick={handleClick}>
              Favorites
              </button>
              <button 
                className='vehicles-btn planet-btn'
                id='vehicles'
                onClick={handleClick}>
              Vehicles
              </button>
            </section>
            <Ship />
          </section>
        )
      }
      export default Home;