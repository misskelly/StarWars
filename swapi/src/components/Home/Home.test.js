import React from 'react';
import Home from './Home';
import { shallow } from 'enzyme';
import FeaturedFilm from '../FeaturedFilm/FeaturedFilm'

// import { fetchFilm } from '../../helpers/fetchCalls.js'
import { mockCleanFilm, mockDirtyFilm } from '../../helpers/mockData.js';

describe('Home', () => {
      let wrapper;
      let mockUpdateActiveComponent;

  beforeEach(() => {
  
    mockUpdateActiveComponent = jest.fn();
  
    wrapper = shallow(<Home updateActiveComponent={ mockUpdateActiveComponent } film={ mockCleanFilm }/>)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
  
  it('renders a FeaturedFilm component with the correct props', () => {

    expect(wrapper.find(FeaturedFilm).prop('film')).toEqual((mockCleanFilm));

});

})