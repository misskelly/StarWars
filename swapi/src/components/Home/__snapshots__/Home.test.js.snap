// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
<section
  className="home"
>
  <section
    className="left-btns"
  >
    <div
      className="moon"
    />
    <button
      className="planets-btn planet-btn"
      id="planets"
      onClick={[Function]}
    >
      Planets
    </button>
    <button
      className="people-btn planet-btn"
      id="people"
      onClick={[Function]}
    >
      People
    </button>
  </section>
  <FeaturedFilm
    film={
      Object {
        "scroll": "Turmoil has engulfed the
Galactic Republic. The taxation
of trade routes to outlying star
systems is in dispute.

Hoping to resolve the matter
with a blockade of deadly
battleships, the greedy Trade
Federation has stopped all
shipping to the small planet
of Naboo.

While the Congress of the
Republic endlessly debates
this alarming chain of events,
the Supreme Chancellor has
secretly dispatched two Jedi
Knights, the guardians of
peace and justice in the
galaxy, to settle the conflict....",
        "subtitle": "The Phantom Menace",
        "title": "Episode I",
        "year": "1999",
      }
    }
    scroll="Turmoil has engulfed the
Galactic Republic. The taxation
of trade routes to outlying star
systems is in dispute.

Hoping to resolve the matter
with a blockade of deadly
battleships, the greedy Trade
Federation has stopped all
shipping to the small planet
of Naboo.

While the Congress of the
Republic endlessly debates
this alarming chain of events,
the Supreme Chancellor has
secretly dispatched two Jedi
Knights, the guardians of
peace and justice in the
galaxy, to settle the conflict...."
    subtitle="The Phantom Menace"
    title="Episode I"
    year="1999"
  />
  <section
    className="right-btns"
  >
    <button
      className="favs-btn planet-btn"
      id="favs"
      onClick={[Function]}
    >
      Favorites
    </button>
    <button
      className="vehicles-btn planet-btn"
      id="vehicles"
      onClick={[Function]}
    >
      Vehicles
    </button>
  </section>
  <Ship />
</section>
`;
